enclave/core/sgx/longjmp.S

```diff
@@ -32,5 +32,11 @@ val_is_nonzero:
     mov 40(%rdi),%r13
     mov 48(%rdi),%r14
     mov 56(%rdi),%r15
-    jmp *16(%rdi)
+
+    // Load indirect jump address into register and issue an lfence to mitigate
+    // LVI.
+    mov 16(%rdi),%rdi
+    lfence
```

Comment:
Assembler-based LVI mitigation is suppose to insert `lfence` before the indirect branch with register operand. No need to manually insert one. For SGX apps targeting CPUs not affected by LVI only, Assembler-based LVI mitigation is not required, nor is manually inserted `lfence`. Stopping speculative execution when its not necessary for security reason does impact performance.

If the assembly sub-routine is not executed frequently, an extra `lfence` inserted manually probably will not cause too much performance degradation though.

Reply:
Agree with @bodzhang. Similar to the case above, the assembler should be able to recognize the converted indirect `jmp` and correctly insert `lfence` for us.

Reply:
Inspecting the generated assembly code,
```bash
objdump -d echo_enc-lvi-cfg | grep -B 2 "jmpq   \*"

--
  129c9e:       4c 8b 77 30             mov    0x30(%rdi),%r14
  129ca2:       4c 8b 7f 38             mov    0x38(%rdi),%r15
  129ca6:       ff 67 10                jmpq   *0x10(%rdi)
```

the assembler fails to insert an lfence here. Therefore, we need to add an lfence here.
@mingweishih Can you please double-check if my analysis is correct?

Reply:
@mingweishih nvm. I double-checked and removed the extra lfence.